songpal/containers.py

```diff
@@ -340,9 +374,10 @@ def __str__(self):
             s += " (active)"
         return s
 
-    async def activate(self):
+    async def activate(self, output: Zone = None):
         """Activate this input."""
-        return await self.services["avContent"]["setPlayContent"](uri=self.uri)
+        return await self.services["avContent"]["setPlayContent"](uri=self.uri, output=output.uri)
```

Comment:
I have not yet tested this in practice, but I would like to make sure this also works with devices without zones before this is merged.

Reply:
Agreed. I don't have a device that doesn't have zones, so hard to mimic what the behavior is there

Reply:
This indeed breaks when there is no zone available:
```
  File "/home/hass/home-assistant/libs/python-sony/songpal/containers.py", line 389, in activate
    return await self.services["avContent"]["setPlayContent"](uri=self.uri, output=output.uri)
AttributeError: 'NoneType' object has no attribute 'uri'
```

I think the easiest fix is to check for Noneness and pass `""` as output, then it works:
```
if output is None:
    output = ""
<call>
```

Reply:
Instead changing it to


```
output_uri = output.uri if output else ""
return await self.services["avContent"]["setPlayContent"](uri=self.uri, output=output_uri)

```